### PR TITLE
Rewrite and simplify unit tests to run under gulp

### PR DESCRIPTION
**Pull request by lmorchard**
Looks like the original tests were aimed more at integration testing the whole stack including findApp(). These tests just focus on the few things the module does

### CODE REVIEW

File: tests/unit/test.index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }).catch(function(err) {
+      test.ifError(err);
+      test.done();
+    });
+  }
+
+};
+
+// Build just enough of a mock client to yield web apps for findApp()
+function buildMockClientWithWebapps(apps) {
```
Comment (tofumatt) [inline tests/unit/test.index.js]: Does it make sense to use the mock library here, just for consistency?

**Comment by tofumatt** [comment]
Looks good to me, though maybe using the same types of mocks everywhere is better? I'll leave that to you, so r+wc

File: tests/unit/test.index.js (hunk 1 above)
Comment (lmorchard) [inline tests/unit/test.index.js]: Well, mockery doesn't really add much here. That's mostly to intercept `require()` calls and swap in a replacement. Here, I can just build a simple object locally and pass it as a parameter.

Comment (tofumatt) [inline tests/unit/test.index.js]: I meant more as a “this is the way we always do mocks, consistent API” type thing. If the code will not look any*more* complicated with mockery, I’d say use it just for consistency’s sake.

Comment (lmorchard) [inline tests/unit/test.index.js]: What I'm saying, though, is mockery doesn't do anything here. There's no module to `require()`. I just need to build a minimal object that responds to what `findApps()` calls

Comment (tofumatt) [inline tests/unit/test.index.js]: Gotcha, merged!

**Pull request merged**